plugins/main/public/components/common/search-bar/search-bar-service.ts

```diff
@@ -1,114 +1,148 @@
+export const search = async (
+  params: SearchParams,
+): Promise<SearchResponse | void> => {
+  const {
+    indexPattern,
+    filters: defaultFilters = [],
+    query,
+    pagination,
+    sorting,
+    fields,
+    aggs,
+  } = params;
+  if (!indexPattern) {
+    return;
+  }
+  const data = getPlugins().data;
+  const searchSource = await data.search.searchSource.create();
+  const fromField =
+    (pagination?.pageIndex || 0) * (pagination?.pageSize || 100);
+  const sortOrder: OpenSearchQuerySortValue[] =
+    sorting?.columns.map(column => {
+      const sortDirection = column.direction === 'asc' ? 'asc' : 'desc';
+      return { [column?.id || '']: sortDirection } as OpenSearchQuerySortValue;
     }) || [];
-    let filters = defaultFilters;
+  let filters = defaultFilters;
 
-    // check if dateRange is defined
-    if (params.dateRange && params.dateRange?.from && params.dateRange?.to) {
-        const { from, to } = params.dateRange;
-        filters = [
-            ...filters,
-            {
-                // @ts-ignore
-                range: {
-                    [indexPattern.timeFieldName || 'timestamp']: {
-                        gte: from,
-                        lte: to,
-                        format: 'strict_date_optional_time'
-                    }
-                }
-            }
-        ]
+  // check if dateRange is defined
+  if (params.dateRange && params.dateRange?.from && params.dateRange?.to) {
+    // when the from and to are equal, the to is now
+    /* ToDo: Search the OSD date range parser, to prevent this change and fix
+        when the fllter is "Today" and "This week". In those ranges the filter to and from are the same
+        Today: from: now/d, to: now/d
+        This Week: from: now/w to:now/w
+    */
+    if (params.dateRange.from === params.dateRange.to) {
+      params.dateRange.to = 'now';
     }
+    const { from, to } = params.dateRange;
 
-    const searchParams = searchSource
-        .setParent(undefined)
-        .setField('filter', filters)
-        .setField('query', query)
-        .setField('sort', sortOrder)
-        .setField('size', pagination?.pageSize)
-        .setField('from', fromField)
-        .setField('index', indexPattern)
+    filters = [
+      ...filters,
+      {
+        // @ts-ignore
+        range: {
+          [indexPattern.timeFieldName || 'timestamp']: {
+            gte: dateMath.parse(from).toISOString(),
+            lte: dateMath.parse(to).toISOString(),
+            format: 'strict_date_optional_time',
+          },
+        },
+      },
+    ];
+  }
 
-    // add fields
-    if (fields && Array.isArray(fields) && fields.length > 0) {
-        searchParams.setField('fields', fields);
-    }
-    try {
-        return await searchParams.fetch();
-    } catch (error) {
-        if (error.body) {
-            throw error.body;
-        }
-        throw error;
+  if (aggs) {
+    searchSource.setField('aggs', aggs);
+  }
```

Comment:
**nitpick**: maybe move this under https://github.com/wazuh/wazuh-dashboard-plugins/pull/6482/files#diff-3dcb0d72ff73b91a2ae78cb2d802bb63ffdbd71d1a85d846587df7ff4af6f4acR82-R84